bindings/java/src/main/java/org/apache/opendal/Operator.java

```diff
@@ -205,6 +208,18 @@ public CompletableFuture<Void> rename(String sourcePath, String targetPath) {
         return AsyncRegistry.take(requestId);
     }
 
+    public CompletableFuture<Void> removeAll(String path) {
+        final long requestId = removeAll(nativeHandle, path);
+        return AsyncRegistry.take(requestId);
+    }
+
+    public CompletableFuture<List<Entry>> list(String path) {
+        final long requestid = list(nativeHandle, path);
+        final CompletableFuture<Entry[]> result = AsyncRegistry.take(requestid);
+        Objects.requireNonNull(result);
+        return result.thenApplyAsync(Arrays::asList);
```

Comment:
```suggestion
        return Objects.requireNonNull(result).thenApplyAsync(Arrays::asList);
```

nit - personal preference